File: Semester_3_Angular_Personal/Admin/src/app/services/product.service.ts
import { Injectable } from "@angular/core";
import { BaseURLService } from "./baseUrl.service";
import { HttpClient } from "@angular/common/http";
import { lastValueFrom } from "rxjs";
import { Product } from "../models/product.model";

@Injectable()
export class ProductService {
    constructor(
        private baseURlService: BaseURLService,
        private httpClient: HttpClient,
    ){}

    async findAll(){
        return await lastValueFrom(this.httpClient.get(this.baseURlService.baseURL() + 'admin/product/findAll'));
    }

    async find(id: string) {
        return await lastValueFrom(this.httpClient.get(this.baseURlService.baseURL() + 'admin/product/find/' + id));
    }

    async find2(id: string) {
        return await lastValueFrom(this.httpClient.get(this.baseURlService.baseURL() + 'admin/product/find2/' + id));
    }

    async create(product: Product) {
        return await lastValueFrom(this.httpClient.post(this.baseURlService.baseURL() + 'admin/product/create', product));
    }

    async edit(product: Product){
        return await lastValueFrom(this.httpClient.put(this.baseURlService.baseURL() + 'admin/product/edit', product));
    }

    async upload(id: string, file: File) {
        var formData = new FormData();
        formData.append('file', file);
        return await lastValueFrom(this.httpClient.post(this.baseURlService.baseURL() + 'admin/product/uploadFile/' + id, formData));
    }
}

File: Semester_3_API_Personal/Semester_3_API_Personal/Service/ProductService.cs
using Semester_3_API_Personal.Models;

namespace Semester_3_API_Personal.Service;

public interface ProductService
{
    public dynamic findAll();
    public dynamic findAll2();
    public dynamic findTwo();
    public dynamic findThree();
    public bool Create(Product product);
    public bool Edit(Product product);
    public bool Delete(int id);
    public dynamic find(int id);
    public dynamic find2(int id);
    public dynamic searchByKeyword(string keyword);
    public dynamic findDetailById(int id);
    public Product findByIdNoTracking(int id);
}
